src/compile/model.ts

```diff
@@ -231,27 +231,34 @@ export abstract class Model {
    * Return the data source name for the given data source type.
    *
    * For unit spec, this is always simply the spec.name + '-' + dataSourceType.
-   * However, for composite specs, data sources might get merged, thus we have to
+   * We already use the name map so that marks and scales use the correct data.
    */
   public dataName(dataSourceType: DataTable): string {
-    return this._dataNameMap.get(this.name(dataSourceType + ''));
+    return this._dataNameMap.get(this.name(String(dataSourceType)));
+  }
+
+  /**
+   * Function to rename the dataset name using the name map. Used in the assemble step.
+   */
+  public dataRename(name: string): string {
+    return this._dataNameMap.get(name);
```

Comment:
This is `get`ting, not rename.   I also have the `renameData` method right above on line 226.